### PR TITLE
Layout improvements in the reports page.

### PR DESCRIPTION
**Pull request by akitaonrails**
- [x] Making the grid have 2 columns instead of just 1.
- [x] Moving the time range links as navigation buttons in the top
- [x] Adding specs for the Iteration Service
- [x] Adding calculations for Velocity Volatility

### CODE REVIEW

File: app/views/projects/reports.html.erb [hunk 1]
```diff
@@ -2,69 +2,78 @@
...
 <div class="row">
   <div class="col-sm-12 col-md-12">
-    <h2><%= t('projects.reports.current_iteration') %> <small>(<%= current_iteration_start_date %> - <%= current_iteration_end_date %>)</small></h2>
 
-    <p>This iteration has <%= current_iteration_points %> points planned in <%= current_iteration_count %> stories (including <%= current_iteration_non_estimable %> bugs or chores).</p>
+    <div class="col-sm-6 col-md-6">
+      <h3><%= t('projects.reports.current_iteration') %> <small>(<%= current_iteration_start_date %> - <%= current_iteration_end_date %>)</small></h3>
 
-    <p>There are <%= accepted_points %> points accepted so far, which is <%= accepted_rate %> of the total.</p>
+      <p>This iteration has <%= current_iteration_points %> points planned in <%= current_iteration_count %> stories (including <%= current_iteration_non_estimable %> bugs or chores).</p>
```
Comment (adbatista) [inline app/views/projects/reports.html.erb]: this should be in i18n right?

File: spec/services/iteration_service_spec.rb [hunk 2]
```diff
@@ -97,4 +97,116 @@
       expect(service.date_for_iteration_number(5)).to eq(Time.zone.parse("2011/11/08"))
     end
   end
+
+  context 'complete set of stories in many different iterations' do
+    let(:today) { Time.zone.parse('2016-08-31') }
+    let(:dummy) { create(:user, username: "dummy", email: "dummy@foo.com", name: "Dummy", initials: "XX")}
+    before do
+      Timecop.freeze(today)
```
Comment (adbatista) [inline spec/services/iteration_service_spec.rb]: just curious, why not use `travel_to` ?

File: app/views/projects/reports.html.erb (hunk 1 above)
Comment (akitaonrails) [inline app/views/projects/reports.html.erb]: Should, but the phrase formation depends on several <%= %> which will be very tricky to compose in all the different languages. This part and the activity presenter are still missing proper I18n for now

File: spec/services/iteration_service_spec.rb (hunk 2 above)
Comment (akitaonrails) [inline spec/services/iteration_service_spec.rb]: Don't remember the difference, but freeze works :-P

**Comment by adbatista** [comment]
:+1:

File: app/helpers/projects_helper.rb [hunk 3]
```diff
@@ -67,14 +67,34 @@ def accepted_points
   end
 
   def accepted_rate
-    number_to_percentage(( accepted_points.to_f * 100.0 ) / current_iteration_points.to_f)
+    if current_iteration_points == 0
```
Comment (dmitryrck) [inline app/helpers/projects_helper.rb]: `current_iteration_points.zero?` :nail_care:

File: app/helpers/projects_helper.rb [hunk 4]
```diff
@@ -67,14 +67,34 @@ def accepted_points
...
+  def last_iteration_start_date(worst = false)
+    iterations = if worst
+                   velocity       = @service.velocity
+                   volatility     = @service.volatility
+                   worst_velocity = velocity - ( velocity * volatility )
+                   @service.backlog_iterations(worst_velocity.floor)
+                 else
+                   @service.backlog_iterations
+                 end
+    iterations.last.start_date.to_date.to_s(:long)
```
Comment (dmitryrck) [inline app/helpers/projects_helper.rb]: :neutral_face:

File: app/views/projects/_project.html.erb [hunk 5]
```diff
@@ -1,7 +1,8 @@
 <li class="list-group-item">
   <h4 class="item-title">
     <%= link_to truncate(project.name, length: 18), project, class: 'tag-tooltip',
-      data: {toggle: "tooltip"}, title: project.name %> <span class="pull-right">(<%= project.stories_count %>)</span>
```
Comment (dmitryrck) [inline app/views/projects/_project.html.erb]: `:tooltip` :nail_care:

File: app/views/projects/reports.html.erb [hunk 6]
```diff
@@ -2,69 +2,78 @@
...
-<% end %>
-
-<% cache [@project, params[:since]], expires: 2.days do %>
-
-<% unless since_options.empty? %>
-<div class="row">
-  <div class="col-sm-12 col-md-12 text-right">
-    |
+  <div class="project_nav">
+    <%= link_to t('Back'), project_path(@project), class: "btn btn-primary btn-sm" %>
```
Comment (dmitryrck) [inline app/views/projects/reports.html.erb]: it is better to put `'Back'` in locale.

File: spec/services/iteration_service_spec.rb [hunk 7]
```diff
@@ -97,4 +97,116 @@
...
+        estimate = story_type == 'bug' ? nil : project.point_values[rand.rand(project.point_values.size)]
+        project.stories.create!(title: "Story #{65 + i}", story_type: story_type, estimate: estimate, requested_by: dummy)
+      end
+    end
+
+    after { Timecop.return }
+
+    it '#group_by_iteration' do
+      groups = service.group_by_iteration
+      expect(groups).to eq({1=>[3, 2, 0, 5, 0],
```
Comment (dmitryrck) [inline spec/services/iteration_service_spec.rb]: I guess you don't need `{` here :P

File: app/services/iteration_service.rb [hunk 8]
```diff
@@ -183,5 +183,26 @@ def current_iteration_details
     end
   end
 
+  def standard_deviation(groups = [], sample = false)
+    # algorithm: https://www.mathsisfun.com/data/standard-deviation-formulas.html
+    #
+    mean            = groups.sum.to_f / groups.size.to_f
+    differences_sqr = groups.map { |velocity| (velocity.to_f - mean) ** 2 }
+    count = sample ? (groups.size - 1).to_f : groups.size.to_f
```
Comment (dmitryrck) [inline app/services/iteration_service.rb]: ```(sample ? (groups.size - 1) : groups.size).to_f``` :nail_care:

**Pull request merged**